python/cudf/cudf_pandas_tests/test_fast_slow_proxy.py

```diff
@@ -545,3 +545,7 @@ def test_tuple_with_attrs_transform():
     assert b == bprime and b is not bprime
     assert c == cprime and c is not cprime
     assert d == dprime and d is not dprime
+
+
+def test_fast_slow_function_call():
```

Comment:
As far as testing goes, I don't see a test for `_fast_slow_function_call` itself. There are other tests in [test_fast_slow_proxy.py](https://github.com/rapidsai/cudf/blob/branch-24.06/python/cudf/cudf_pandas_tests/test_fast_slow_proxy.py) for `test_fast_slow_arg_*`. Should I add a test `_fast_slow_function_call` itself?

Also, the plan is for `_fast_slow_function_call` to return a warning if the results from cudf and pandas differ. What's the best way to handle this in a test?

Reply:
Ideally to test this feature we want to test this at a higher level with something that will definitely differ between pandas and cudf and intentionally fail. Maybe something like

```python
def test_...():
    def udf(x):
        return np.random.random(1).item()

    with set_option("mode.pandas_debugging", True):
        with pytest.raises(...):
            xdf.Series([1]).apply(udf)
```

Reply:
Yup. Here's the docs on testing that warnings were raised: https://docs.pytest.org/en/7.1.x/how-to/capture-warnings.html#warns